dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -244,6 +258,109 @@ public Experiment end(String experimentId, User user)
         return factory.save(ended);
     }
 
+    @WrapInTransaction
+    @Override
+    public Experiment addVariant(final String experimentId, final User user)
+            throws DotDataException, DotSecurityException {
+
+        final Experiment persistedExperiment = find(experimentId, user)
+                .orElseThrow(()->new DoesNotExistException("Experiment with provided id not found"));
+
+        final String variantNameBase = EXPERIMENT_VARIANT_NAME_PREFIX + experimentId
+                + EXPERIMENT_VARIANT_NAME_SUFFIX;
+
+        final int nextAvailableIndex = getNextAvailableIndex(variantNameBase);
+
+        final String variantName = variantNameBase + nextAvailableIndex;
+
+        final String variantDescription = EXPERIMENT_VARIANT_DESCRIPTION
+                + nextAvailableIndex;
+
+        variantAPI.save(Variant.builder().name(variantName).name(variantName).build());
+
+        final ExperimentVariant experimentVariant = ExperimentVariant.builder().id(variantName)
+                .description(variantDescription).weight(0).build();
+
+        final TrafficProportion trafficProportion = persistedExperiment.trafficProportion();
+
+        final TreeSet<ExperimentVariant> variants = new TreeSet<>();
+        variants.addAll(trafficProportion.variants());
+        variants.add(experimentVariant);
+
+        // IF split evenly - re-distribute weights
```

Comment:
move to a local method better instead of write a comment

Reply:
Was already done